src/Microsoft.AspNet.Mvc.Core/Formatters/OutputFormatter.cs

```diff
@@ -273,5 +271,63 @@ private Encoding MatchAcceptCharacterEncoding(IList<StringWithQualityHeaderValue
 
             return null;
         }
+
+        // There's no allocation-free way to sort an IList and we may have to filter anyway,
+        // so we're going to have to live with the copy + insertion sort.
+        private IList<StringWithQualityHeaderValue> Sort(IList<StringWithQualityHeaderValue> values)
```

Comment:
Worth creating a `IList<T> Sort<T>(IList<T>, IComparer<T>)` method and sharing that?